src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_ExplicitQuery_IntArrayParam_Optional.generated.txt

```diff
@@ -0,0 +1,442 @@
+                options.EndpointBuilder.Metadata.Add(new ProducesResponseTypeMetadata(statusCode: StatusCodes.Status200OK, type: typeof(global::System.Int32[]), contentTypes: GeneratedMetadataConstants.JsonContentType));
+                return new RequestDelegateMetadataResult { EndpointMetadata = options.EndpointBuilder.Metadata.AsReadOnly() };
+            };
+            RequestDelegateFactoryFunc createRequestDelegate = (del, options, inferredMetadataResult) =>
+            {
+                Debug.Assert(options != null, "RequestDelegateFactoryOptions not found.");
+                Debug.Assert(options.EndpointBuilder != null, "EndpointBuilder not found.");
+                Debug.Assert(options.EndpointBuilder.ApplicationServices != null, "ApplicationServices not found.");
+                Debug.Assert(options.EndpointBuilder.FilterFactories != null, "FilterFactories not found.");
+                var handler = Cast(del, global::System.Int32[] (global::System.Int32[]? arg0= default) => throw null!);
+                EndpointFilterDelegate? filteredInvocation = null;
+                var serviceProvider = options.ServiceProvider ?? options.EndpointBuilder.ApplicationServices;
+                var logOrThrowExceptionHelper = new LogOrThrowExceptionHelper(serviceProvider, options);
+                var jsonOptions = serviceProvider?.GetService<IOptions<JsonOptions>>()?.Value ?? FallbackJsonOptions;
+                var jsonSerializerOptions = jsonOptions.SerializerOptions;
+                jsonSerializerOptions.MakeReadOnly();
+                var objectJsonTypeInfo = (JsonTypeInfo<object?>)jsonSerializerOptions.GetTypeInfo(typeof(object));
+                var responseJsonTypeInfo =  (JsonTypeInfo<global::System.Int32[]?>)jsonSerializerOptions.GetTypeInfo(typeof(global::System.Int32[]));
+
+                if (options.EndpointBuilder.FilterFactories.Count > 0)
+                {
+                    filteredInvocation = GeneratedRouteBuilderExtensionsCore.BuildFilterDelegate(ic =>
+                    {
+                        if (ic.HttpContext.Response.StatusCode == 400)
+                        {
+                            return ValueTask.FromResult<object?>(Results.Empty);
+                        }
+                        return ValueTask.FromResult<object?>(handler(ic.GetArgument<global::System.Int32[]?>(0)!));
+                    },
+                    options.EndpointBuilder,
+                    handler.Method);
+                }
+
+                Task RequestHandler(HttpContext httpContext)
+                {
+                    var wasParamCheckFailure = false;
+                    // Endpoint Parameter: p (Type = int[]?, IsOptional = True, IsParsable = True, IsArray = True, Source = Query)
+                    var p_raw = httpContext.Request.Query["p"];
+                    var p_temp = p_raw.ToArray();
+                    global::System.Int32[]? p_local = new global::System.Int32[p_temp.Length];
```

Comment:
```suggestion
                    global::System.Int32[]? p_local = p_temp.Length > 0 ? new global::System.Int32[p_temp.Length] : null;
```

Or maybe fallback to the default value defined for the parameter if no query strings are provided?

In this case, if no query strings are provided you'd get an empty array as input to the handler and not the default value which I don't think is the desired behavior.

Reply:
> In this case, if no query strings are provided you'd get an empty array as input to the handler and not the default value which I don't think is the desired behavior.

At first I implemented lazy array creation inside loop on first parameter, but the tests failed because srcgen parameter was null, but runtime bound was always an array instance.

I didn't dig deep into runtime binding implementation, but it seems that it never binds null array even if no parameters are present.
Therefore I removed the lazy array instantiation to keep the same behaviour between both modes.

If the different behaviour is acceptable, I will bring back lazy array init.